controllers/state_of_the_world.go

```diff
@@ -0,0 +1,170 @@
+		{Kind: ptr.To(machinery.GatewayGroupKind), EventType: ptr.To(controller.UpdateEvent)},
+		{Kind: ptr.To(machinery.HTTPRouteGroupKind)},
+		{Kind: ptr.To(kuadrantv1beta2.AuthPolicyKind)},
+	}
+
+	istioGatewayProvider := &IstioGatewayProvider{Client: client}
+	effectivePolicesReconciler.ReconcileFuncs = append(effectivePolicesReconciler.ReconcileFuncs, (&controller.Subscription{
+		ReconcileFunc: istioGatewayProvider.Sample,
+		Events:        append(commonAuthPolicyResourceEventMatchers, controller.ResourceEventMatcher{Kind: ptr.To(IstioAuthorizationPolicyKind)}),
+	}).Reconcile)
+	effectivePolicesReconciler.ReconcileFuncs = append(effectivePolicesReconciler.ReconcileFuncs, (&controller.Subscription{
+		ReconcileFunc: istioGatewayProvider.Sample,
+		Events: []controller.ResourceEventMatcher{
+			{Kind: ptr.To(machinery.GatewayGroupKind), EventType: ptr.To(controller.DeleteEvent)},
+		},
+	}).Reconcile)
+
+	reconciler := &controller.Workflow{
+		Precondition: func(ctx context.Context, resourceEvents []controller.ResourceEvent, _ *machinery.Topology) {
+			logger := controller.LoggerFromContext(ctx).WithName("event logger")
+			for _, event := range resourceEvents {
+				// log the event
+				obj := event.OldObject
+				if obj == nil {
+					obj = event.NewObject
+				}
+				values := []any{
+					"type", event.EventType.String(),
+					"kind", obj.GetObjectKind().GroupVersionKind().Kind,
+					"namespace", obj.GetNamespace(),
+					"name", obj.GetName(),
+				}
+				if event.EventType == controller.UpdateEvent && logger.V(1).Enabled() {
+					values = append(values, "diff", cmp.Diff(event.OldObject, event.NewObject))
+				}
+				logger.Info("new event", values...)
+			}
+		},
+		Tasks: []controller.ReconcileFunc{
+			topologyFileReconciler.Reconcile,
+			effectivePolicesReconciler.Reconcile,
+		},
+	}
+
+	return reconciler.Run
+}
+
+type EffectivePolicesReconciler struct {
+	Client         *dynamic.DynamicClient
+	ReconcileFuncs []controller.ReconcileFunc
+}
+
+func (r *EffectivePolicesReconciler) Reconcile(ctx context.Context, _ []controller.ResourceEvent, _ *machinery.Topology) {
+	logger := controller.LoggerFromContext(ctx).WithName("effective polices reconciler")
+	logger.Info("reconciling effective polices reconciler")
+}
+
+type TopologyFileReconciler struct {
+	Client *dynamic.DynamicClient
+}
+
+func (r *TopologyFileReconciler) Reconcile(ctx context.Context, _ []controller.ResourceEvent, topology *machinery.Topology) {
+	logger := controller.LoggerFromContext(ctx).WithName("topology file")
+	cm := &corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "topology",
+			Namespace: "kuadrant-system",
+		},
+		Data: map[string]string{
+			"topology": topology.ToDot(),
+		},
+	}
+
+	configMapRes := schema.GroupVersionResource{Group: "", Version: "v1", Resource: "configmaps"}
+	unstructuredMap, err := runtime.DefaultUnstructuredConverter.ToUnstructured(cm)
```

Comment:
Can you use [`controller.Destruct`](https://github.com/Kuadrant/policy-machinery/blob/e963418c9566a5e61acdd766c6109d3f0c147222/controller/runnable.go#L213) here maybe?

```suggestion
	unstructuredMap, _ := controller.Destruct(cm)
```

Reply:
I have done this and dropped the checking of the error. I am not sure how I feel about not doing something with the errors that may come back.